ee/clickhouse/migrations/0014_persons_distinct_ids_column_remove.py

```diff
@@ -0,0 +1,12 @@
+from infi.clickhouse_orm import migrations
+
+from ee.clickhouse.sql.person import KAFKA_PERSONS_TABLE_SQL, PERSONS_TABLE_MV_SQL
+from posthog.settings import CLICKHOUSE_CLUSTER
+
+operations = [
+    migrations.RunSQL(f"DROP TABLE person_mv ON CLUSTER {CLICKHOUSE_CLUSTER}"),
+    migrations.RunSQL(f"DROP TABLE kafka_person ON CLUSTER {CLICKHOUSE_CLUSTER}"),
```

Comment:
What happens to rows in this table when you drop it? (in other words, do we end up dropping messages when we run this migration?)

Reply:
This is OK, we have done this many times already.

So this table does not have any "rows" in a normal sense. Instead this is a kafka engine table, which is a shim for a kafka consumer for a particular topic. 

When you read from it it reads data from the topic and saves the offsets in the kafka consumer group. If you drop it, nothing happens to the consumer group - just nothing is reading from it. When we create the table again it re-connects and once the mv table exists it starts reading from kafka.

Reply:
Yup exactly right.
https://clickhouse.tech/docs/en/sql-reference/statements/create/view/#materialized
https://clickhouse.tech/docs/en/engines/table-engines/integrations/kafka/

This is just a view pulling from that kafka table (wrapping a kafka topic). Both are ephemeral and don't actually persist any rows. By dropping them you are basically making the cluster forget about the topic in kafka, but the consumer group is still hanging around. As long as you recreate the kafka table with the same consumer group it will start pulling from the exact same offset as before. No data loss! 🎉 As for the materialized view that's just the driving force pulling data from kafka and dumping it into the downstream table that does the actual persisting.

Reply:
Thanks for the links! :)